### PR TITLE
add documentation for timezone preference

### PR DESCRIPTION
**Pull request by taimoorzaeem**
Documentation for timezone preference.

### CODE REVIEW

File: docs/references/api/preferences.rst [hunk 1]
```diff
@@ -45,3 +45,50 @@ The server ignores unrecognized or unfulfillable preferences by default. You can
...
+    {"t":"2023-10-18T07:37:59.611-07:00"},
+    {"t":"2023-10-18T09:37:59.611-07:00"}
+  ]
+
+For an invalid timezone, PostgREST returns values with the default timezone. However, with ``handling=strict``, an invalid timezone preference will throw an error.
+
+.. code-block:: http
+
+  GET /timestamps HTTP/1.1
+  Prefer: timezone=Jupiter/Red_Spot
```
Comment (steve-chavez) [inline docs/references/api/preferences.rst:80]: :ringed_planet: :red_circle:
```suggestion
  Prefer: handling=strict, timezone=Jupiter/Red_Spot
```

File: docs/references/api/preferences.rst [hunk 2]
```diff
@@ -45,3 +45,50 @@ The server ignores unrecognized or unfulfillable preferences by default. You can
...
+  HTTP/1.1 200 OK
+  Content-Type: application/json; charset=utf-8
+
+.. code-block:: json
+  
+  [
+    {"t":"2023-10-18T12:37:59.611+00:00"},
+    {"t":"2023-10-18T14:37:59.611+00:00"},
+    {"t":"2023-10-18T16:37:59.611+00:00"}
+  ]
```
Comment (steve-chavez) [inline docs/references/api/preferences.rst:93]: Hm, there should be an error here

Comment (steve-chavez) [reply docs/references/api/preferences.rst:93]: Ah, I see you meant to show the `lenient` way here.

**Comment by laurenceisla** [comment]
The working with data types how-to mentions this:

https://github.com/PostgREST/postgrest-docs/blob/084e8c2950d4bd843ae1fcd40b39510a18e435b4/docs/how-tos/working-with-postgresql-data-types.rst?plain=1#L65

We could add a link to this section there:

```rst
 The response has the date in the time zone configured by the server: ``UTC -05:00`` (see :ref:`pref_timezone`).
```

File: docs/references/api/preferences.rst [hunk 3]
```diff
@@ -45,3 +45,50 @@ The server ignores unrecognized or unfulfillable preferences by default. You can
...
+
```
Comment (steve-chavez) [inline docs/references/api/preferences.rst:94]: Some links and elaboration
```suggestion
For an invalid timezone, PostgREST returns values with the default timezone (configured on``postgresql.conf`` or as a setting on the :ref:`authenticator <roles>`).

.. code-block:: http

  GET /timestamps HTTP/1.1
  Prefer: timezone=Jupiter/Red_Spot

.. code-block:: http

  HTTP/1.1 200 OK
  Content-Type: application/json; charset=utf-8

.. code-block:: json
  
  [
    {"t":"2023-10-18T12:37:59.611+00:00"},
    {"t":"2023-10-18T14:37:59.611+00:00"},
    {"t":"2023-10-18T16:37:59.611+00:00"}
  ]

Note that there's no ``Preference-Applied`` in the response.

However, with ``handling=strict``, an invalid timezone preference will throw an :ref:`error <pgrst122>`.

.. code-block:: http

  GET /timestamps HTTP/1.1
  Prefer: timezone=Jupiter/Red_Spot
  
 .. code-block:: http

  HTTP/1.1 400 Bad Request
```

**Pull request merged**